File: src/envLegacy.js
/**
 * Read a config env var with a legacy fallback name (donor → initiator migration).
 * @param {string} primary
 * @param {string} legacy
 * @returns {string | undefined}
 */
export function readEnvWithLegacy(primary, legacy) {
  const primaryRaw = process.env[primary];
  if (typeof primaryRaw === "string" && primaryRaw.trim()) {
    return primaryRaw.trim();
  }
  const legacyRaw = process.env[legacy];
  if (typeof legacyRaw === "string" && legacyRaw.trim()) {
    return legacyRaw.trim();
  }
  return undefined;
}

File: src/main/java/org/sharingbridge/integration/config/EnvLegacy.java
package org.sharingbridge.integration.config;

/** Read a config env var with a legacy fallback name (donor → initiator). */
public final class EnvLegacy {

    private EnvLegacy() {}

    public static String readEnvWithLegacy(String primary, String legacy) {
        String primaryRaw = System.getenv(primary);
        if (primaryRaw != null && !primaryRaw.trim().isEmpty()) {
            return primaryRaw.trim();
        }
        String legacyRaw = System.getenv(legacy);
        if (legacyRaw != null && !legacyRaw.trim().isEmpty()) {
            return legacyRaw.trim();
        }
        return null;
    }
}
